xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/delete.vm

```diff
@@ -238,20 +238,20 @@
 #######################################################
 #macro(displayConfirmMessage)
   #if($xwiki.hasRecycleBin())
-    #if($services.refactoring.isAllowedToSkipTheRecycleBin())
-      <div class="xform">
+    #if($services.refactoring.isRecycleBinSkippingAllowed())
+      <div class="xform" id="skipRecycleBinForm">
         <dl>
           <dt>
-            $escapetool.xml($services.localization.render('core.recyclebin.wantToSkip'))
+            $escapetool.xml($services.localization.render('core.recycleBin.shouldSkipRecycleBin'))
           </dt>
           <dd>
             <label class="xwiki-form-listclass">
-              <input name="skipRecycleBin" type="radio" value="false" checked/>
-              $escapetool.xml($services.localization.render('core.recyclebin.wantToSkip.no'))
+              <input name="shouldSkipRecycleBin" type="radio" value="false" checked/>
+              $escapetool.xml($services.localization.render('core.recycleBin.shouldSkipRecycleBin.no'))
```

Comment:
Why repeat recycleBin? 

`core.recycleBin.shouldSkip.no` is much better and follows the conventions which explicitly mention this:

> for example, a key starting with theme.colors.wizard. refers to a text located in the wizard for the color submodule of the theme application, macro.python.parameter. refers to parameters of the python macro,

Reply:
I agree. I was following the way translation keys are defined in the administration, but the third part of the key has to match the field id for technical reasons. This is not the case here.
Do you prefer `recycleBin` or `recyclebin` for the second part of the keys?

Reply:
> Do you prefer recycleBin or recyclebin for the second part of the keys?

After more thinking, I think it's best to use the agreed convention, ie use the uppercase, even if it doesn't follow what was started. In theory all the existing keys should be deprecated to use that but it's too much a pain so let's leave them as they are now.